### PR TITLE
Omit metrics artifacts (alpha) from the bill of materials (bom)

### LINKED ISSUE #2268: Remove the metrics modules from the BOM
We don't want non-stable signals to be in the otel java BOM. It will be required for users to manually add them as dependencies and keep the versions in sync themselves.

### PR DESCRIPTION
**Pull request by breedx-splk**
This resolves #2268.

### CODE REVIEW

File: bom/build.gradle [hunk 1]
```diff
@@ -11,6 +11,7 @@ dependencies {
         parent.childProjects.sort { "$it.value" }
                 .collect { it.value }
                 .findAll { it.name != project.name }
+                .findAll { !it.name.endsWith('-metrics') }
```
Comment (anuraaga) [inline bom/build.gradle:14]: Does `it.version.endsWith('-alpha')` work?

Comment (jkwatson) [reply bom/build.gradle:14]: maybe `.contains`, since SNAPSHOT always ends up last.

Comment (breedx-splk) [reply bom/build.gradle:14]: It doesn't work.  It seems to be the case that the dependency resolution happens early enough that the version has not been customized in the submodule yet...or at least not consistently/reliably.

**Review by anuraaga — APPROVED** [review]

File: bom/build.gradle (hunk 1 above)
Comment (anuraaga) [reply bom/build.gradle:14]: Too bad - thanks for checking!

**Comment by breedx-splk** [comment]
Since this was already covered by #2329 I am going to close this out.  Thanks for looking!

**Pull request closed without merging**